src/payment/strategies/checkoutcom/checkoutcom-google-pay-payment-strategy.ts

```diff
@@ -0,0 +1,109 @@
+import { InternalCheckoutSelectors } from '../../../checkout';
+import { InvalidArgumentError } from '../../../common/error/errors';
+import { ContentType } from '../../../common/http-request/';
+import { OrderRequestBody } from '../../../order';
+import Payment from '../../payment';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import { GooglePayPaymentStrategy, PaymentMethodData } from '../googlepay';
+
+import { CheckoutcomGooglePayToken, CheckoutcomToken } from './checkout';
+
+export default class CheckoutcomGooglePayPaymentStrategy extends GooglePayPaymentStrategy {
+    TEST_URL: string = 'https://api.sandbox.checkout.com/tokens';
+    LIVE_URL: string = 'https://api.checkout.com/tokens';
+
+    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        if (!this._googlePayOptions) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.googlepaycheckoutcom" argument is not provided.');
+        }
+
+        const { payment: { methodId = '' } = {} } = payload;
+        const { paymentMethods: { getPaymentMethodOrThrow } } = this._store.getState();
+
+        const testMode: boolean = !!getPaymentMethodOrThrow(methodId).config.testMode;
+        const { checkoutcomkey } = getPaymentMethodOrThrow(methodId).initializationData;
+
+        const {
+            onError = () => {},
+            onPaymentSelect = () => {},
+        } = this._googlePayOptions;
+
+        return Promise.resolve(this._getPayment())
+            .then(payment => {
+                if (!payment.paymentData.nonce || !payment.paymentData.cardInformation) {
+
+                    return this._googlePayPaymentProcessor.displayWallet()
+                        .then(paymentData => this._paymentInstrumentSelected(paymentData))
+                        .then(() => onPaymentSelect())
+                        .then(() => this._getPayment())
+                        .catch(error => {
+                            if (error.statusCode !== 'CANCELED') {
+                                onError(error);
+                            }
+                        });
+                }
+
+            })
+            .then(() => this._convert_token(testMode, checkoutcomkey))
+            .then(payment => {
+                const paymentData: Payment = payment as Payment;
+
+                return this._store.dispatch(this._orderActionCreator.submitOrder({useStoreCredit: payload.useStoreCredit}, options))
+                    .then(() => this._store.dispatch(this._paymentActionCreator.submitPayment(paymentData)));
+            });
+    }
+
+    async requestCheckoutcomTokenize(testMode: boolean | undefined, checkoutcomKey: string, data = {}): Promise<CheckoutcomToken> {
```

Comment:
Also, this seems to be a private method so it should be marked with `private` and have an underscore prefix.

Reply:
@davidchin this is because, i couldn't find a way to mock global.fetch, so instead i mock this method so i a make it public, if i could use requestSender i think i could mock it properly, but the issue with the header it adds make it not viable right know i think